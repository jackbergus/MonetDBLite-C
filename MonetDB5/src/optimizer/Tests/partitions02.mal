#testing the BAT partition manager
# use a fake sql library
# see how you handle an aggregation
module sql;
pattern bind(s:str,t:str,i:str,flag:int):bat[:oid,:int]
address MDBdummy;
pattern rsColumn(b:any...):void
address MDBdummy;
pattern resultSet(b:any...):void
address MDBdummy;
pattern exportResult(b:any...):void
address MDBdummy;
pattern exportValue(b:any...):void
address MDBdummy;

#select count(*) from x
function query(A0:sht):void;
    x1:bat[:oid,:int]{notnil=true,rows=0,bid=765}  := sql.bind("sys","x","i",0);
    x6:bat[:oid,:int]{notnil=true,rows=0,bid=769}  := sql.bind("sys","x","i",1);
    constraints.emptySet(x6);
    x8:bat[:oid,:int]{notnil=true,rows=0,bid=771}  := sql.bind("sys","x","i",2);
    constraints.emptySet(x8);
    x11{rows=0} := algebra.markT(x1,0@0);
    x12{rows=0} := bat.reverse(x11);
    x13{rows=1} := aggr.count(x12);
    sql.exportValue(1,"sys.","countx","int",32,0,6,x13,"");
end query;
optimizer.partitions("sql","query");
mdb.list("sql","query");
