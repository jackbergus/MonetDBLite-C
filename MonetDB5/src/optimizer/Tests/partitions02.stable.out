stdout of test 'partitions02` in directory 'src/optimizer` itself:


# 22:44:54 >  
# 22:44:54 >  Mtimeout -timeout 60 mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs" --set "xquery_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/xquery_logs" --set mapi_port=39818 --set xrpc_port=40069 --set monet_prompt= --trace --dbname=mTests_src_optimizer  partitions02.mal
# 22:44:54 >  

# MonetDB Server v5.0.0_beta2_1
# Copyright (c) 1993-2007 CWI, all rights reserved
# Compiled for x86_64-redhat-linux-gnu/64bit with 64bit OIDs dynamically linked
# dbname:mTests_src_optimizer
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://eir.ins.cwi.nl:32348/
pattern sql.bind(s:str,t:str,i:str,flag:int):bat[:oid,:int] address MDBdummy;
pattern sql.rsColumn(b:any...):void address MDBdummy;
pattern sql.resultSet(b:any...):void address MDBdummy;
pattern sql.exportResult(b:any...):void address MDBdummy;
pattern sql.exportValue(b:any...):void address MDBdummy;
function sql.query(A0:sht):void;
    x1 := sql.bind("sys","x","i",0);
    x6 := sql.bind("sys","x","i",1);
    constraints.emptySet(x6);
    x8 := sql.bind("sys","x","i",2);
    constraints.emptySet(x8);
    x11 := algebra.markT(x1,0@0);
    x12 := bat.reverse(x11);
    x13 := aggr.count(x12);
    sql.exportValue(1,"sys.","countx","int",32,0,6,x13,"");
end query;
function user.main():void;
#testing the BAT partition manager 
# use a fake sql library 
# see how you handle an aggregation 
#select count(*) from x 
    mdb.List("sql","query");
end main;
function sql.query(A0:sht):void;        #  0 query:void := sql.query(A0:sht) {G}
    x1:bat[:oid,:int] {notnil=true,rows=0,bid=765} := sql.bind("sys","x","i",0);#  1 x1:bat[:oid,:int] := MDBdummy(_6:str, _7:str, _8:str, _4:int) {G}
    x6:bat[:oid,:int] {notnil=true,rows=0,bid=769} := sql.bind("sys","x","i",1);#  2 x6:bat[:oid,:int] := MDBdummy(_6:str, _7:str, _8:str, _11:int) {G}
    bpm9:bat[:oid,:int]  := bpm.unfold(x6{notnil=true,rows=0,bid=769});#  3 bpm9:bat[:oid,:int] := BPMunfold(x6:bat[:oid,:int]) {G}
    bpm.emptySet(bpm9);                 #  4 _12:void := BPMemptySet(bpm9:bat[:oid,:int]) 
    x8:bat[:oid,:int] {notnil=true,rows=0,bid=771} := sql.bind("sys","x","i",2);#  5 x8:bat[:oid,:int] := MDBdummy(_6:str, _7:str, _8:str, _15:int) {G}
    bpm13:bat[:oid,:int]  := bpm.unfold(x8{notnil=true,rows=0,bid=771});#  6 bpm13:bat[:oid,:int] := BPMunfold(x8:bat[:oid,:int]) {G}
    bpm.emptySet(bpm13);                #  7 _16:void := BPMemptySet(bpm13:bat[:oid,:int]) 
    x11{rows=0} := algebra.markT(x1{notnil=true,rows=0,bid=765},0@0);#  8 x11:bat[:oid,:oid] := ALGtmark(x1:bat[:oid,:int], _18:oid) {G}
    x12{rows=0} := bat.reverse(x11{rows=0});#  9 x12:bat[:oid,:oid] := BKCreverse(x11:bat[:oid,:oid]) {G}
    x13{rows=1} := aggr.count(x12{rows=0});# 10 x13:wrd := ALGcount_bat(x12:bat[:oid,:oid]) 
    sql.exportValue(1,"sys.","countx","int",32,0,6,x13{rows=1},"");# 11 _21:void := MDBdummy(_11:int, _22:str, _23:str, _24:str, _25:int, _4:int, _26:int, x13:wrd, _27:str) 
catch SQLexception;                     # 12 SQLexception:str {J16}
    bpm.garbage(bpm9);                  # 13 _34:void := BPMgarbage(bpm9:bat[:oid,:int]) 
    bpm.garbage(bpm13);                 # 14 _35:void := BPMgarbage(bpm13:bat[:oid,:int]) 
    raise SQLexception;                 # 15 SQLexception:str 
exit SQLexception;                      # 16 SQLexception:str 
    bpm.garbage(bpm9);                  # 17 _38:void := BPMgarbage(bpm9:bat[:oid,:int]) 
    bpm.garbage(bpm13);                 # 18 _39:void := BPMgarbage(bpm13:bat[:oid,:int]) 
end query;                              # 19  

# 22:44:54 >  
# 22:44:54 >  Done.
# 22:44:54 >  

